blog/_posts/2016-05-17-automated-data-validation.md

```diff
@@ -0,0 +1,136 @@
+the data is invalid.  If a user adds "bad" data, the "build" fails and
+issues a report indicating what went wrong.
+
+[![Data CI](/img/posts/data_ci_travis.png)](https://github.com/frictionlessdata/ex-continuous-data-integration)
+
+As an example, this CSV is "bad".
+
+```
+Date,Country,Number
+2015-01-01,3,20.3
+2015-02-01,United States,23.5
+2015-02,United States,x23.5
+```
+
+When we packaged it (see its
+[datapackage.json](https://github.com/frictionlessdata/ex-continuous-data-integration/blob/master/datapackage.json)),
+we gave it the following JSON Table Schema.
+
+{% highlight json %}
+{
+  "fields": [
+    {
+      "name": "Date",
+      "type": "date",
+      "description": ""
+    },
+    {
+      "name": "Country",
+      "type": "string",
+      "description": ""
+    },
+    {
+      "name": "Number",
+      "type": "number",
+      "description": ""
+    }
+  ]
+}
+{% endhighlight %}
+
+On commit, this data would result in the following report:
+
+```
++----------------+-------------------------------------------------------------+
+| result_name    | result_message                                              |
++================+=============================================================+
+| Incorrect Type | The value "2015-02" in column "Date" is not a valid Date.   |
++----------------+-------------------------------------------------------------+
+| Incorrect Type | The value "x23.5" in column "Number" is not a valid Number. |
++----------------+-------------------------------------------------------------+
+```
+
+## How It Works
+
+The Data Package descriptor file,
+[datapackage.json](http://dataprotocols.org/data-packages/), provides
+both high-level metadata as well as a
+[schema](http://frictionlessdata.io/guides/json-table-schema/) for
+tabular data.  The Python libary
+[datapackage-py](http://github.com/frictionlessdata/datapackage-py)
+provides a high-level model of the Data Package that allows us to
+inspect and work with the data inside.  We call out to the
+[GoodTables API](http://goodtables.okfnlabs.org/api) service
+([previously blogged about on this channel](http://okfnlabs.org/blog/2015/03/06/goodtables-web-service.html))
+to validate our tabular data.  Given that our example relies on
+[GitHub as a data storage mechanism](http://blog.okfn.org/2013/07/02/git-and-github-for-data/),
+[Travis CI](http://travis-ci.org/) served as a convenient host to test
+out this idea, but this approach is broadly applicable to any storage
+backend with some extra tweaking (e.g. using AWS
+[Lambda](https://aws.amazon.com/lambda/) and
+[S3](https://aws.amazon.com/s3/)).
+
+
+On every update, two small test functions read the `datapackage.json`
+to read and validate the tabular data contained therein according to
+its structure and adherence to a
+[schema](http://frictionlessdata.io/guides/json-table-schema/).
+Structural errors include whether the CSV is malformed in some way
```

Comment:
More about GoodTables and validation ... - could we not just say "read the guide on Frictionless Data site about validating data"